composer.json

```diff
@@ -20,15 +20,24 @@
             "homepage": "https://github.com/friendsofsymfony/FOSHttpCache/contributors"
         }
     ],
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/ddeboer/guzzle6-adapter"
+        }
+    ],
     "require": {
-        "php": ">=5.3.3",
-        "guzzle/guzzle": "~3.8",
+        "php": ">=5.4.8",
         "symfony/event-dispatcher": "~2.3",
-        "symfony/options-resolver": "~2.3"
+        "symfony/options-resolver": "~2.3",
+        "php-http/adapter": "dev-internal_separation",
+        "guzzlehttp/psr7": "~1.0"
```

Comment:
You don't need to use it to create custom requests. Ideally, custom request should also be implementation agnostic (at least they will be in php-http, see my comment bellow, how). I will show you an example about custom messages once I finished the client.